### PR TITLE
[Issue 1035] Allow layer regulazier set after model definition, add regulazier for cadd and cmul

### PR DESCRIPTION
**Pull request by SeaOfOcean**
## What changes were proposed in this pull request?

Allow layer regulazier set after model definition
## How was this patch tested?
unit test

## Related links or issues (optional)
fix https://github.com/intel-analytics/BigDL/issues/1035

### CODE REVIEW

**Comment by SeaOfOcean** [comment]
pr validate pass

**Comment by psyyz10** [comment]
LGTM

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/torch/CAddSpec.scala [hunk 1]
```diff
@@ -16,13 +16,82 @@
...
 import com.intel.analytics.bigdl.tensor.Tensor
 import com.intel.analytics.bigdl.utils.RandomGenerator._
+import com.intel.analytics.bigdl.utils.T
 
 @com.intel.analytics.bigdl.tags.Serial
 class CAddSpec extends TorchSpec {
-    "A CAdd(5, 1)" should "generate correct output and grad" in {
+  "CAdd L2 regularizer" should "works correctly" in {
+    import com.intel.analytics.bigdl.numeric.NumericDouble
+    torchCheck()
```
Comment (qiuxin2012) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/torch/CAddSpec.scala]: Don't need torchCheck for this UT.

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/torch/CMulSpec.scala [hunk 2]
```diff
@@ -16,13 +16,82 @@
...
 import com.intel.analytics.bigdl.tensor.Tensor
 import com.intel.analytics.bigdl.utils.RandomGenerator._
+import com.intel.analytics.bigdl.utils.T
 
 @com.intel.analytics.bigdl.tags.Serial
 class CMulSpec extends TorchSpec {
-    "A CMul(5, 1)" should "generate correct output and grad" in {
+  "CMul L2 regularizer" should "works correctly" in {
+    import com.intel.analytics.bigdl.numeric.NumericDouble
+    torchCheck()
```
Comment (qiuxin2012) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/torch/CMulSpec.scala]: Don't need torchCheck for this UT.

**Comment by zhichao-li** [comment]
We should  add python wrapper as well while adding it into scala.  
related PR: https://github.com/intel-analytics/BigDL/pull/1008

**Comment by zhichao-li** [comment]
Should we also add regulazier to this class: ```VolumetricConvolution``` as well?

**Comment by zhichao-li** [comment]
LGTM

**Pull request merged**